src/io.rs

```diff
@@ -717,7 +717,7 @@ pub trait ReadBytesExt: io::Read {
     /// ```
     #[inline]
     fn read_i8_into(&mut self, dst: &mut [i8]) -> Result<()> {
-        let buf = unsafe { slice_to_u8_mut(dst) };
+        let buf = unsafe { &mut *(dst as *mut [i8] as *mut [u8]) };
```

Comment:
Sorry, but could you please say more about why you made this change?

Reply:
Looking back on it now, I don't see a big motivation -- it removes a redundant length and slice reconstruction in debug builds, but the explicitness of `slice_to_u8_mut` is fine and gets optimized away in release builds.

`write_i8_into` is the reason for this PR, so I can remove this commit.

Reply:
Yeah that would be great, thanks.